crates/solver/src/settlement_ranker.rs

```diff
@@ -196,17 +192,65 @@ impl SettlementRanker {
                 )),
             );
         }
-        for (i, (solver, _, _)) in enumerate(&rated_settlements) {
-            let rank = rated_settlements.len() - i;
-            solver.notify_auction_result(auction_id, AuctionResult::Ranked(rank));
-            self.metrics
-                .settlement_simulation(solver.name(), SolverSimulationOutcome::Success);
-        }
 
+        // Before sorting, make sure to shuffle the settlements. This is to make sure we
+        // don't give preference to any specific solver when there is an
+        // objective value tie.
+        rated_settlements.shuffle(&mut rand::thread_rng());
+
+        if self.enable_auction_rewards {
+            rated_settlements.sort_by(|a, b| a.1.score.score().cmp(&b.1.score.score()));
+
+            rated_settlements.iter_mut().rev().enumerate().for_each(
+                |(i, (solver, settlement, _))| {
+                    self.metrics
+                        .settlement_simulation(solver.name(), SolverSimulationOutcome::Success);
+                    settlement.ranking = i + 1;
+                    solver.notify_auction_result(auction_id, AuctionResult::Ranked(i + 1));
+                },
+            );
+        } else {
+            // TODO: remove this block of code once `auction-rewards` is implemented
+            rated_settlements.sort_by(|a, b| compare_solutions(&a.1, &b.1, self.decimal_cutoff));
+
+            let auction_based_ranking = auction_based_ranking(
+                rated_settlements
+                    .iter()
+                    .map(|(_, settlement, _)| settlement)
+                    .collect::<Vec<_>>(),
+            );
+
+            rated_settlements.iter_mut().rev().enumerate().for_each(
+                |(i, (solver, settlement, _))| {
+                    self.metrics
+                        .settlement_simulation(solver.name(), SolverSimulationOutcome::Success);
+                    settlement.ranking = auction_based_ranking
+                        .get(&settlement.id)
+                        .copied()
+                        .unwrap_or(0);
+                    solver.notify_auction_result(auction_id, AuctionResult::Ranked(i + 1));
+                },
+            );
+        }
         Ok((rated_settlements, errors))
     }
 }
 
+// TODO: remove this once `auction-rewards` is implemented
+// Sort settlements by `auction-rewards` rules and return hashmap of settlement
+// id to ranking
+fn auction_based_ranking(settlements: Vec<&RatedSettlement>) -> HashMap<usize, usize> {
+    let mut settlements = settlements;
```

Comment:
```suggestion
fn auction_based_ranking(mut settlements: Vec<&RatedSettlement>) -> HashMap<usize, usize> {
```

Reply:
Will keep this one since it's more clear that we are doing thing on a copy. But this function will be removed anyway.